wire/msgtx.go

```diff
@@ -255,6 +291,30 @@ func (msg *MsgTx) BtcDecode(r io.Reader, pver uint32) error {
 		return err
 	}
 
+	// A count of zero (meaning no TxIn's to the uninitiated) indicates
+	// this is a transaction with witness data.
+	var flag [1]byte
+	if count == 0 && pver == WitnessVersion {
+		// Next, we need to read the flag, which is a single byte.
+		if _, err = r.Read(flag[:]); err != nil {
```

Comment:
This needs to be `io.ReadFull(r, flag[:])`.  Regular `r.Read(flag[:])` does **not**  enforce the byte to be read if it's not ready.  That condition is also not an error (since the number of bytes read will simply be 0) and therefore this code, which is not checking the number of read bytes, could end up failing incorrectly when it checks the flag below and finds it to be 0x00 instead of 0x01 due to it not actually being read.